File: src/XmlParser.Core/Attributes/ColumnNameAttribute.cs
namespace XmlParser.Core.Attributes;

/// <summary>
/// An attribute for assigning class properties to Excel column.
/// </summary>
[AttributeUsage(AttributeTargets.Property)]
public sealed class ColumnNameAttribute(string name) : Attribute
{
    /// <summary>
    /// Name of the column in the Excel file.
    /// </summary>
    public string Name => name;
}

File: src/XlsxParser.Core.FSharp/Attributes/ColumnNameAttribute.fs
namespace XlsxParser.Core.FSharp.Attributes

open System

/// An attribute for assigning class properties to Excel column.
[<Sealed>]
type ColumnNameAttribute(name: string) =
    inherit Attribute()
    /// Name of the column in the Excel file.
    member this.Name with get () = name
